metadata-ingestion/tests/unit/grafana/test_grafana_entity_mcp_builder.py

```diff
@@ -0,0 +1,226 @@
+import pytest
+
+from datahub.ingestion.source.grafana.entity_mcp_builder import (
+    _build_custom_properties,
+    _build_dashboard_properties,
+    _build_ownership,
+    build_chart_mcps,
+    build_dashboard_mcps,
+)
+from datahub.ingestion.source.grafana.models import Dashboard, Panel
+from datahub.metadata.schema_classes import (
+    ChartInfoClass,
+    DashboardInfoClass,
+    DataPlatformInstanceClass,
+    GlobalTagsClass,
+    OwnershipClass,
+    StatusClass,
+)
+
+
+@pytest.fixture
+def mock_panel():
+    return Panel(
+        id="1",
+        title="Test Panel",
+        description="Test Description",
+        type="graph",
+        targets=[{"query": "SELECT * FROM test"}],
+        datasource={"type": "mysql", "uid": "test_uid"},
+    )
+
+
+@pytest.fixture
+def mock_dashboard():
+    return Dashboard(
+        uid="dash1",
+        title="Test Dashboard",
+        description="Test Description",
+        version="1",
+        panels=[],
+        tags=["tag1", "environment:prod"],
+        timezone="UTC",
+        schemaVersion="1.0",
+        meta={"folderId": "123"},
+        created_by="test@test.com",
+    )
+
+
+def test_build_custom_properties():
+    panel = Panel(
+        id="1",
+        title="Test Panel",
+        type="graph",
+        description="Test Description",
+        targets=[{"query": "test"}],
+        datasource={"type": "mysql", "uid": "test_uid"},
+    )
+
+    props = _build_custom_properties(panel)
+    assert props["type"] == "graph"
+    assert props["datasourceType"] == "mysql"
+    assert props["datasourceUid"] == "test_uid"
+    assert props["description"] == "Test Description"
+    assert props["queryCount"] == "1"
+
+
+def test_build_dashboard_properties(mock_dashboard):
+    props = _build_dashboard_properties(mock_dashboard)
+    assert props["version"] == "1"
+    assert props["schema_version"] == "1.0"
+    assert props["timezone"] == "UTC"
+
+
+def test_build_ownership(mock_dashboard):
+    ownership = _build_ownership(mock_dashboard)
+    assert isinstance(ownership, OwnershipClass)
+    assert len(ownership.owners) == 2
+    assert any(owner.owner.split(":")[-1] == "dash1" for owner in ownership.owners)
+    assert any(owner.owner.split(":")[-1] == "test" for owner in ownership.owners)
```

Comment:
this is the same, right? I find proposal here easier to read

```
    assert any(owner.owner == "urn:li:copuser:dash1" for owner in ownership.owners)
    assert any(owner.owner == "urn:li:copuser:test" for owner in ownership.owners)
```

or even, something like

```
    assert set([owner.owner for owner in ownership.owners]) == set(
        "urn:li:corpuser:dash1", 
        "urn:li:corpuser:test",
    )
```